src/app.ts

```diff
@@ -3,19 +3,57 @@ import cors from "cors";
 import { getStatus } from "./get-status";
 import startTransfersMonitor from './transfers';
 import { log } from './debug';
+import { setExchangeStatus } from './setExchangeStatus';
 
 const app = express();
 const port = process.env.PORT || 8081;
 
+if(process.env.PASSWORD === undefined) {
+    throw new Error("Missing PASSWORD in .env!");
+}
+
 startTransfersMonitor();
 
 app.use(cors());
+app.use(express.json());
+
 app.get("/", async (req, res) => {
   let status = await getStatus();
   res.setHeader("Content-Type", "application/json");
   res.send(status);
 });
 
+app.post("/status", async (req, res) => {
+  const password = req.header("x-api-key");
+
+  if (process.env.PASSWORD !== password) {
+    res.status(401).send({
+      message: "Wrong authentication credentials!"
+    });
+    return;
+  }
+
+  const { exchangeIndex, exchangeStatus } = req.body;
+
+  if (!exchangeIndex || !exchangeStatus) {
```

Comment:
This will fail if `exchangeIndex === 0`, which is a valid index.
```suggestion
  if (exchangeIndex === undefined || exchangeStatus === undefined)
```